test/e2e/vsphere_utils.go

```diff
@@ -0,0 +1,266 @@
+func getVSpherePersistentVolumeClaimSpec(namespace string, labels map[string]string) *v1.PersistentVolumeClaim {
+	var (
+		pvc *v1.PersistentVolumeClaim
+	)
+	pvc = &v1.PersistentVolumeClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "pvc-",
+			Namespace:    namespace,
+		},
+		Spec: v1.PersistentVolumeClaimSpec{
+			AccessModes: []v1.PersistentVolumeAccessMode{
+				v1.ReadWriteOnce,
+			},
+			Resources: v1.ResourceRequirements{
+				Requests: v1.ResourceList{
+					v1.ResourceName(v1.ResourceStorage): resource.MustParse("2Gi"),
+				},
+			},
+		},
+	}
+	if labels != nil {
+		pvc.Spec.Selector = &metav1.LabelSelector{MatchLabels: labels}
+	}
+
+	return pvc
+}
+
+// function to create vmdk volume
+func createVSphereVolume(vsp *vsphere.VSphere, volumeOptions *vsphere.VolumeOptions) (volumePath string, err error) {
+	if volumeOptions == nil {
+		volumeOptions = new(vsphere.VolumeOptions)
+		volumeOptions.CapacityKB = 2097152
+		volumeOptions.Name = "e2e-vmdk-" + strconv.FormatInt(time.Now().UnixNano(), 10)
+	}
+	volumePath, err = vsp.CreateVolume(volumeOptions)
+	Expect(err).NotTo(HaveOccurred())
+	return volumePath, nil
+}
+
+// function to write content to the volume backed by given PVC
+func writeContentToVSpherePV(client clientset.Interface, ns string, pvc *v1.PersistentVolumeClaim, content string) {
+	podClient := client.CoreV1().Pods(ns)
+	name := "writerpod" + strconv.FormatInt(time.Now().UnixNano(), 10)
+	framework.Logf("Creating POD: %s to write content to volume", name)
+
+	writerPod := &v1.Pod{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Pod",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		Spec: v1.PodSpec{
+			Containers: []v1.Container{
+				{
+					Name:    name,
+					Image:   "gcr.io/google_containers/busybox:1.24",
+					Command: []string{"/bin/sh"},
+					Args:    []string{"-c", "echo '" + content + "' > /mnt/file.txt && chmod o+rX /mnt /mnt/file.txt"},
+					VolumeMounts: []v1.VolumeMount{
+						{
+							Name:      pvc.Name,
+							MountPath: "/mnt",
+						},
+					},
+				},
+			},
+			RestartPolicy: v1.RestartPolicyNever,
+			Volumes: []v1.Volume{
+				{
+					Name: pvc.Name,
+					VolumeSource: v1.VolumeSource{
+						PersistentVolumeClaim: &v1.PersistentVolumeClaimVolumeSource{
+							ClaimName: pvc.Name,
+						},
+					},
+				},
+			},
+		},
+	}
+
+	defer func() {
+		podClient.Delete(name, nil)
+	}()
+
+	writerPod, err := podClient.Create(writerPod)
+	framework.ExpectNoError(err, "Failed to create write pod: %v", err)
+	err = framework.WaitForPodSuccessInNamespace(client, writerPod.Name, writerPod.Namespace)
+	Expect(err).NotTo(HaveOccurred())
+	framework.Logf("Done with writing content to volume")
+}
+
+// function to verify content is matching on the volume backed for given PVC
+func verifyContentOfVSpherePV(client clientset.Interface, ns string, pvc *v1.PersistentVolumeClaim, expectedContent string) {
```

Comment:
Defining more generic reader and writer pods, passing in the command to be executed, seems useful. Maybe these can be defined in _pvutil.go_?